test/utils/Arrays.test.js

```diff
@@ -4,22 +4,22 @@ const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
 
 const { randomArray, generators } = require('../helpers/random');
 
-// See https://en.cppreference.com/w/cpp/algorithm/ranges/lower_bound
+// See https://en.cppreference.com/w/cpp/algorithm/lower_bound
 const lowerBound = (array, value) => {
   const i = array.findIndex(element => value <= element);
```

Comment:
According to the C++ Reference:

> Returns an iterator pointing to the first element in the range [first, last) such that element < value (or comp(element, value)) is false, (i.e. that is greater than or equal to value), or last if no such element is found.

What I see in this function is that `findIndex` returns the first element that satisfies the `value <= element` condition, but according to C++'s, shouldn't it be `value > element`? Seems off to me.

```suggestion
  const i = array.findIndex(element => value > element);
```

Reply:
Solving, I see this is wrong because it's expecting the opposite of `value > element`